client/remoterepo.c

```diff
@@ -106,11 +215,412 @@ _handle_curl_cb(
     }
     return dwError;
 }
+
+
+uint32_t
+TDNF_Check_Digest(
+    const char *filename,
+    hash_op *hash,
+    uint8_t *result
+    )
+{
+    uint32_t dwError = 0;
+    int fd = -1;
+    hash_ctx_t ctx;
```

Comment:
I think we should memset ctx to 0 before passing it's address. Just a good practice

Reply:
this is a pointer which will be allocated by sha/md5 init function. So not required to memset this.

Reply:
please init all variables. @sshedi is right.

Reply:
Added memset for the union variable :)